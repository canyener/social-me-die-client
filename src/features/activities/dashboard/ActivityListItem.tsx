import React from 'react'
import { Item, Button, Segment, Icon } from 'semantic-ui-react'
import { Link } from 'react-router-dom'

import { IActivity } from '../../../app/models/activity'

const ActivityListItem: React.FC<{ activity: IActivity }> = ({ activity }) => {
  return (
    <Segment.Group>
      <Segment>
        <Item.Group>
          <Item>
            <Item.Image size='tiny' circular src='/assets/user.png' />
            <Item.Content>
              <Item.Header as='a'>{activity.title}</Item.Header>
              <Item.Description>
                Hosted by Cancan
            </Item.Description>
            </Item.Content>
          </Item>
        </Item.Group>
      </Segment>
      <Segment>
        <Icon name='clock' /> {activity.date!.toLocaleDateString('tr-TR')}
        <Icon name='marker' /> {activity.venue}, {activity.city}
      </Segment>
      <Segment secondary>
        Attendees will go here
      </Segment>
      <Segment clearing>
        <span>{activity.description}</span>
        <Button
          as={Link}
          to={`/activities/${activity.id}`}
          floated='right'
          content='View'
          color='blue'
        />
      </Segment>
    </Segment.Group>
  )
}

export default ActivityListItem