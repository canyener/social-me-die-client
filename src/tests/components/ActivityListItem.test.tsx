import { shallow } from 'enzyme'

import ActivityListItem from '../../features/activities/dashboard/ActivityListItem'
import { fakeActivities } from '../fixtures/seed'

test('Should render ActivityListItem correctly', () => {
  const wrapper = shallow(<ActivityListItem activity={fakeActivities[0]} />)

  expect(wrapper).toMatchSnapshot()
})