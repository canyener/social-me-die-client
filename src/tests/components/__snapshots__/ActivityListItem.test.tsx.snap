// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render ActivityListItem correctly 1`] = `
<SegmentGroup>
  <Segment>
    <ItemGroup>
      <Item>
        <ItemImage
          circular={true}
          size="tiny"
          src="/assets/user.png"
        />
        <ItemContent>
          <ItemHeader
            as="a"
          >
            Fake Activity 1
          </ItemHeader>
          <ItemDescription>
            Hosted by Cancan
          </ItemDescription>
        </ItemContent>
      </Item>
    </ItemGroup>
  </Segment>
  <Segment>
    <Icon
      as="i"
      name="clock"
    />
     
    1:55 AM
    <Icon
      as="i"
      name="marker"
    />
     
    Fake Venue
    , 
    Fake City
  </Segment>
  <Segment
    secondary={true}
  >
    Attendees will go here
  </Segment>
  <Segment
    clearing={true}
  >
    <span>
      Fake description
    </span>
    <Button
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Link",
          "propTypes": Object {
            "innerRef": [Function],
            "onClick": [Function],
            "replace": [Function],
            "target": [Function],
            "to": [Function],
          },
          "render": [Function],
        }
      }
      color="blue"
      content="View"
      floated="right"
      to="/activities/1"
    />
  </Segment>
</SegmentGroup>
`;
